### PR TITLE
Marker are now updated if user changes map name, when user changes map image markers can be kept or deleted

### PR DESCRIPTION
**Pull request by MaksymShokin**
@tonytlwu @squallstar 

## Issue
Fliplet/fliplet-studio#4193
Fliplet/fliplet-studio#4194
Fliplet/fliplet-studio#5407

## Description
When map name gets changed marker located on that map change.
When the user changes map image Confirm modal fired. If the user selects to delete markers they will be deleted from this map, if the user keeps markers they will be updated according to the difference in images resolution.

## Screenshots/screencasts
66049503-2bf44680-e534-11e9-897e-8ad8ac5cd3a8

## Backward compatibility
This change is fully backward compatible.

## Notes
Made new PR because this PR https://github.com/Fliplet/fliplet-widget-interactive-map/pull/64 has too many conflicts.

### CODE REVIEW

File: js/interface/map-panel.js [hunk 1]
```diff
@@ -60,7 +134,29 @@ Fliplet.InteractiveMap.component('map-panel', {
         }
       })
 
-      window.filePickerProvider.then((result) => {
+      window.filePickerProvider.then(result => {
+       if (this.shouldKeepMarkers) {
```
Comment (squallstar) [inline js/interface/map-panel.js]:
```
if (!this.shouldKeepMarkers) {
  return;
}

// rest of code
```

File: js/interface/map-panel.js [hunk 2]
```diff
@@ -35,6 +77,38 @@ Fliplet.InteractiveMap.component('map-panel', {
       }
     },
     openMapPicker() {
+      Fliplet.DataSources.connect(this.dataSourceId).then(connection => {
+        this.dataSourceConnection = connection
+        connection.find({where: {['Map name']: this.name}}).then(records => {
+          if (records.length) {
```
Comment (squallstar) [inline js/interface/map-panel.js]:
```
if (!records.length) {
  return;
}

// rest of code
```

File: js/interface/map-panel.js [hunk 3]
```diff
@@ -26,7 +26,49 @@ Fliplet.InteractiveMap.component('map-panel', {
...
+     this.oldMapName = this.name
+   },
+   updateDataSource() {
+     Fliplet.DataSources.connect(this.dataSourceId).then(connection => {
+       this.dataSourceConnection = connection
+       connection.find({where: {['Map name']: this.oldMapName}}).then(records => {
+         if (!records.length) {
+           return
+         }
+         this.dataSourceConnection.find().then(records => {
```
Comment (squallstar) [inline js/interface/map-panel.js]: add empty line after `}`

File: js/interface/map-panel.js [hunk 4]
```diff
@@ -26,7 +26,49 @@ Fliplet.InteractiveMap.component('map-panel', {
...
+            records.forEach((elem, index, array) => {
+              if (elem.data['Map name'] === this.oldMapName) {
+                array[index].data['Map name'] = this.name
+              }
+            })
+            this.entries = records
```
Comment (squallstar) [inline js/interface/map-panel.js]: add empty line after `})`

File: js/interface/map-panel.js (hunk 3 above)
Comment (MaksymShokin) [reply js/interface/map-panel.js]: fixed

File: js/interface/map-panel.js (hunk 4 above)
Comment (MaksymShokin) [reply js/interface/map-panel.js]: fixed

File: js/interface/map-panel.js (hunk 2 above)
Comment (MaksymShokin) [reply js/interface/map-panel.js]: fixed

File: js/interface/map-panel.js (hunk 1 above)
Comment (MaksymShokin) [reply js/interface/map-panel.js]: Can't do it because part of the code needs to be executed anyway

**Pull request merged**